stock_inventory_analytic/tests/test_stock_inventory_analytic.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2019 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import SavepointCase
+
+
+class TestInventoryAnalytic(SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestInventoryAnalytic, cls).setUpClass()
+
+        # MODELS
+        cls.product_product_model = cls.env['product.product']
+        cls.product_category_model = cls.env['product.category']
+        cls.wizard_model = cls.env['stock.change.product.qty']
+
+        # INSTANCES
+        cls.category = cls.product_category_model.create({
+            'name': 'Physical (test)',
+            'property_cost_method': 'standard',
+            'property_valuation': 'real_time',
+        })
+        cls.analytic_account = cls.env.ref(
+            'analytic.analytic_agrolait')
+
+    def _create_product(self, name):
+        return self.product_product_model.create({
+            'name': name,
+            'categ_id': self.category.id,
+            'type': 'product',
+            'standard_price': 100, })
+
+    def _product_change_qty(self, product, new_qty,
+                            analytic_account_id=None):
+        values = {
+            'product_id': product.id,
+            'new_quantity': new_qty,
+        }
+        if analytic_account_id:
+            values.update({'analytic_account_id': analytic_account_id.id})
+        wizard = self.wizard_model.create(values)
+        wizard.change_product_qty()
+
+    def test_product_change_qty_analytic(self):
+        product = self._create_product('product_product')
+
+        self._product_change_qty(product, 10, self.analytic_account)
+
+        analytic_line = self.env['account.analytic.line'].search(
```

Comment:
@hveficent IMHO, you have to make a search before calling product_change_qty, take the recordset before, then call a second search and make 'recordset_before - recordset_after' to ensure the analytic line is well created.

Reply:
It makes more sense